### PR TITLE
Add an adjust pitch setting to ModTimeRamp

### LINKED ISSUE #7699: Nightcore + Wind Up = Pain
**Describe the bug:** 
This seems very similar to #3189, but I'm opening a new issue anyways because this one involves the Wind Up mod.
When playing with Nightcore (with Speed increase at 2.0x) and Wind Up (Final Rate 2.0x), you get audio that's... very crispy? It's quite bad in gameplay.

**Screenshots or videos showing encountered issue:** 
It also causes this very nice circle visualizer:
Screenshot from 2020-02-01 00-19-09

[Video Gameplay: https://streamable.com/bnu8x](https://streamable.com/bnu8x) (Bug occurs approx. 1:04 in video)

**osu!lazer version:** 
2020.130.0

**Logs:**

### PR DESCRIPTION
**Pull request by H2n9**
Gives WindUp/WindDown mods a toggle to enable/disable pitch shifting with speed change.
Track preview is also updated on setting toggle.

Helps make combinations of nightcore with time ramp mods more playable.

Resolves #7699

### CODE REVIEW

File: osu.Game/Rulesets/Mods/ModTimeRamp.cs [hunk 1]
```diff
@@ -44,12 +47,15 @@ protected ModTimeRamp()
...
+
+            AdjustPitch.BindValueChanged(updatePitchAdjustment, false);
         }
 
         public void ApplyToTrack(Track track)
         {
             this.track = track;
-            track.AddAdjustment(AdjustableProperty.Frequency, SpeedChange);
+
+            track.AddAdjustment(AdjustPitch.Value ? AdjustableProperty.Frequency : AdjustableProperty.Tempo, SpeedChange);
```
Comment (bdach) [inline osu.Game/Rulesets/Mods/ModTimeRamp.cs:58]: Why isn't this done with a `.TriggerChange()`, the same way `FinalRate` is?

File: osu.Game/Rulesets/Mods/ModTimeRamp.cs [hunk 2]
```diff
@@ -26,6 +26,9 @@ public abstract class ModTimeRamp : Mod, IUpdatableByPlayfield, IApplicableToBea
         [SettingSource("Final rate", "The final speed to ramp to")]
         public abstract BindableNumber<double> FinalRate { get; }
 
+        [SettingSource("Adjust Pitch", "Should pitch be adjusted with speed")]
```
Comment (bdach) [inline osu.Game/Rulesets/Mods/ModTimeRamp.cs:29]: The casing of the setting isn't consistent with the other names (although setting name casing is all over the place right now and probably needs to be consolidated at some point).

File: osu.Game/Rulesets/Mods/ModTimeRamp.cs [hunk 3]
```diff
@@ -75,5 +81,13 @@ public virtual void Update(Playfield playfield)
         /// <param name="amount">The amount of adjustment to apply (from 0..1).</param>
         private void applyAdjustment(double amount) =>
             SpeedChange.Value = InitialRate.Value + (FinalRate.Value - InitialRate.Value) * Math.Clamp(amount, 0, 1);
+
+        private void updatePitchAdjustment(ValueChangedEvent<bool> value)
+        {
+            // remove existing old adjustment
+            track.RemoveAdjustment(value.OldValue ? AdjustableProperty.Frequency : AdjustableProperty.Tempo, SpeedChange);
+
+            track.AddAdjustment(value.NewValue ? AdjustableProperty.Frequency : AdjustableProperty.Tempo, SpeedChange);
```
Comment (bdach) [inline osu.Game/Rulesets/Mods/ModTimeRamp.cs:90]: Maybe worth to factor out a local helper?

```csharp
AdjustableProperty adjustmentForPitchSetting(bool value)
    => value ? AdjustableProperty.Frequency : AdjustableProperty.Tempo;
```

**Review by bdach — COMMENTED** [review]
Thanks for the contribution!

I have some notes to the code, but they're not too major.

**Comment by bdach** [comment]
I've made some menial changes, overall I think this is good to go 👍

**Review by smoogipoo — APPROVED** [review]
LGTM, requires an osu-web change which I've made in the referencing PR.

**Pull request merged**